debian/cloud-init.maintscript

```diff
@@ -7,3 +7,4 @@ rm_conffile /etc/init/cloud-init-nonet.conf 0.7.9-243-ge74d775-0ubuntu2~
 rm_conffile /etc/init/cloud-init.conf 0.7.9-243-ge74d775-0ubuntu2~
 rm_conffile /etc/init/cloud-log-shutdown.conf 0.7.9-243-ge74d775-0ubuntu2~
 rm_conffile /etc/NetworkManager/dispatcher.d/hook-network-manager 23.3~4ga31370af-0ubuntu1~
+rm_conffile /etc/cloud/clean.d/README 23.4.1-0ubuntu1~23.10.2~
```

Comment:
Since the commit to include dropping README until our next SRU to a stable release this series won't see the appropriate commit 24.1. We can make all these stable releases match something like the following:

```suggestion
rm_conffile /etc/cloud/clean.d/README 24.1~
```

Also please include an UNRELEASED debian/chagelog entry noting the maintscript changes (there are examples in the debian/changelog for other maintscript changes.

Reply:
We specifically don't want to match against `23.4.1.....2~` because we could have a hot fix of something unrelated which would then require us to bump to 23.4.1-0ubuntu1~23.10.2 on mantic which would then trigger the drop of the README unintentionally